031-custom_authentication.md

```diff
@@ -0,0 +1,104 @@
+# Custom Authentication
+
+This proposal focuses on supporting custom authentication in the Strimzi operator.
+
+## Current situation
+
+Currently, numerous authentication methods are [supported](https://github.com/strimzi/strimzi-kafka-operator/tree/0.25.0/api/src/main/java/io/strimzi/api/kafka/model/authentication) in Strimzi, and allows the user to configure which one to use. In addition, in the case a custom *authorizer* needs to be used, this is also currently [supported](https://github.com/strimzi/strimzi-kafka-operator/blob/0.25.0/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java#L541-L549). 
+
+However, there is no ability to specify custom authentication, which is what this proposal will focus on.
+
+## Motivation
+
+Supporting custom authentication allows greater flexibility for users of Strimzi who may have bespoke/proprietary authentication requirements. This would greatly broaden user adoption of the operator, as it would also support more common authentication schemes, such as [Amazon MSK](https://docs.aws.amazon.com/msk/latest/developerguide/security_iam_service-with-iam.html), and attract users who have much more stringent requirements (i.e. proprietary auth). 
```

Comment:
Hey @scholzj  sorry for the bad link, I was actually thinking of this: https://github.com/aws/aws-msk-iam-auth#configuring-a-kafka-client-to-use-aws-iam rather than docs/ideas about IAM configuraiton. I.e., how can we enable MSK-based auth on the Kafka-cluster itself.